### PR TITLE
Update AVStream / Camera samples for RS4.

### PR DESCRIPTION
**Pull request by JoelCorley**
Misc. changes/improvements to avscamera and sampledevicemft samples.  
Build (project file) fix for samplemft0.

### CODE REVIEW

File: avstream/sampledevicemft/multipinmft.cpp [hunk 1]
```diff
@@ -156,6 +157,9 @@ STDMETHODIMP CMultipinMft::QueryInterface(
 
 --*/
 
+// This Sample will create a grayscale for known media types. Please remove MF_DEVICEMFT_ADD_GRAYSCALER_ to remove the grayscaler
+// This sample also has photo confirmation enabled remove DMF_DEVICEMFT_PHTOTOCONFIRMATION to remove photo confirmation
```
Comment (saredd) [inline avstream/sampledevicemft/multipinmft.cpp]: Sample is capitalized in one line but not the other

File: avstream/sampledevicemft/multipinmft.cpp [hunk 2]
```diff
@@ -681,6 +685,8 @@ STDMETHODIMP  CMultipinMft::ProcessInput(
     DMFTCHECKHR_GOTO( inPin->SendSample( pSample ), done );
 done:
     DMFTRACE( DMFT_GENERAL, TRACE_LEVEL_INFORMATION, "%!FUNC! exiting %x = %!HRESULT!", hr, hr );
+    // Workaround A Bug in the pipeline which is incorrectly not releasing the sample
```
Comment (saredd) [inline avstream/sampledevicemft/multipinmft.cpp]: Sad that this has to go out in a sample

File: avstream/sampledevicemft/multipinmft.cpp [hunk 3]
```diff
@@ -723,22 +729,24 @@ output pins and populate the corresponding MFT_OUTPUT_DATA_BUFFER with the sampl
             pdwStatus ) ) )
         {
             gotOne = true;
-            // Do photo confirmation if enabled
+            // Do photo confirmation if enabled off the preview stream only
```
Comment (saredd) [inline avstream/sampledevicemft/multipinmft.cpp]: "of" or "from" rather than "off"

File: avstream/sampledevicemft/multipinmft.cpp [hunk 4]
```diff
@@ -723,22 +729,24 @@ output pins and populate the corresponding MFT_OUTPUT_DATA_BUFFER with the sampl
...
 #if defined (MF_DEVICEMFT_PHTOTOCONFIRMATION)
-            BOOL pIsImagePin = FALSE;
+            BOOL pIsPreviewPin = FALSE;
             if (pOutputSamples[i].pSample &&
                 IsPhotoConfirmationEnabled() &&
-                (SUCCEEDED(CheckImagePin(static_cast<IMFAttributes*>(poPin), &pIsImagePin)) && pIsImagePin))
+                ((SUCCEEDED(CheckPreviewPin(static_cast<IMFAttributes*>(poPin), &pIsPreviewPin)) && pIsPreviewPin) && 
+                    InterlockedCompareExchange(reinterpret_cast<PLONG>(&m_firePhotoConfirmation),FALSE,TRUE)))
```
Comment (saredd) [inline avstream/sampledevicemft/multipinmft.cpp]: Are there normally not spaces between the args to interlocked exchange?

File: avstream/sampledevicemft/multipinmft.cpp (hunk 1 above)
Comment (JoelCorley) [reply avstream/sampledevicemft/multipinmft.cpp]: Fixed.

File: avstream/sampledevicemft/multipinmft.cpp (hunk 2 above)
Comment (JoelCorley) [reply avstream/sampledevicemft/multipinmft.cpp]: Won't fix.

File: avstream/sampledevicemft/multipinmft.cpp (hunk 3 above)
Comment (JoelCorley) [reply avstream/sampledevicemft/multipinmft.cpp]: Fixed as "from".

File: avstream/sampledevicemft/multipinmft.cpp (hunk 4 above)
Comment (JoelCorley) [reply avstream/sampledevicemft/multipinmft.cpp]: Probably different editors were used.  Adding spaces.

**Comment by JoelCorley** [comment]
Since Sanjeev's comments were about comment wording and whitespace, I fixed them as part of the merge conflict resolution.

**Pull request merged**